### PR TITLE
Issue 52 - Cannot use Auth API for authentication only

### PR DESCRIPTION
**Pull request by bmacauley**
- added some logic in ApiClient.py to remove 'Authorization':  header when using Authhetication API for authentication only

- added a couple of tests to AuthClientTest.py to test this...

$ nosetests ./tests/AuthClientTest.py
....
----------------------------------------------------------------------
Ran 4 tests in 0.001s

OK

### CODE REVIEW

**Comment by rdegges** [comment]
@jmelberg-okta or @lboyette-okta wanna take a look since you two are working on this? oO

**Comment by joshm91** [comment]
I'd love to see this fix!

File: okta/framework/ApiClient.py [hunk 1]
```diff
@@ -24,16 +24,23 @@ def __init__(self, *args, **kwargs):
         elif len(args) > 1 and args[1]:
             self.api_token = args[1]
         else:
-            raise ValueError('An api_token must be provied to create an ApiClient')
+            # raise ValueError('An api_token must be provied to create an ApiClient')
```
Comment (lboyette-okta) [inline okta/framework/ApiClient.py]: We should remove this `else` condition entirely

File: okta/framework/ApiClient.py [hunk 2]
```diff
@@ -24,16 +24,23 @@ def __init__(self, *args, **kwargs):
...
 
         self.api_version = 1
         self.max_attempts = 4
 
-        self.headers = {
-            'Accept': 'application/json',
-            'Content-Type': 'application/json',
-            'Authorization': 'SSWS ' + self.api_token
-        }
+        if kwargs['pathname'] == '/api/v1/authn':
```
Comment (lboyette-okta) [inline okta/framework/ApiClient.py]: This only matches `/api/v1/authn`, so any other nested authn apis (like enrolling in a factor with `/api/v1/authn/factors`) would attempt to send the `api_token`

File: tests/AuthClientTest.py [hunk 3]
```diff
@@ -11,3 +11,10 @@ def tests_client_initializer_args(self):
 
     def tests_client_initializer_kwargs(self):
         client = AuthClient(base_url='https://example.okta.com', api_token='api_key')
+
+
+    def tests_client_initializer_args_2(self):
```
Comment (lboyette-okta) [inline tests/AuthClientTest.py]: 🎆 thanks for writing tests!

**Comment by bretterer** [comment]
Thank you for your PR and sorry it took so long to get back to it. We have recently released a 1.x version of this SDK which completely updates the structure and build process for our SDK.

This SDK is meant to be used for resource management and not authentication purposes. If you believe I am mis-understanding this, please let me know. For now, I am going to close this PR since we have fully re-built this SDK

**Pull request closed without merging**